### DISCUSSION TITLE
Tiddlywiki performance by Lighthouse

### DISCUSSION THREAD

**Discussion by kookma** [TiddlyWiki/TiddlyWiki5#5578] [General]
Using the Lighthouse in Chrome DevTools, I tried to check the performance of tiddlywiki.com and I got the below result

![image](https://user-images.githubusercontent.com/830394/113243972-4b0a9a80-92c9-11eb-884f-95a002fe5fe9.png)

I am wondering to see if there is a simple not time consuming way to improve the performance specially for empty.html

**Comment by abesamma** [comment]
Not surprising. TW5 isn't expected to work like lightning fast and performant static page websites or an SPAs. It's pretty much the antithesis of what most web developers would expect (big blob of JavaScript with an unorthodox thing called wikistore, hence the question mark in the best practices score 😀). But the key thing is that, in edit mode, it behaves like a decent sized app which isn't expected to be "performant", but I think performance can be improved gradually with subsequent versions.

**Comment by pmario** [comment]
>I am wondering to see if there is a simple not time consuming way to improve the performance specially for empty.html

There are some hints, that are probably easy to solve. Especially at tiddlywiki.com ... eg: alt-texts for images, ... So to improve accessibility, which imo is important anyway. 

Increase the size of "tappable" elements, may be possible. ... BUT TW is a PC app atm. ... 

We could add some elements, that are needed for a PWA, to satisfy lighthouse. ... Which may give us a better score, but probably won't affect the overall behaviour. .. Just make the test happy!

I think it should be easy to satisfy "best practices" 

On my PC I can see something readable at tiddlywiki.com after about 1.5 seconds. Lighthouse thinks it's 11 seconds, which is complete nonsense. ... It probably is just an artefact, how they measure it. 

One of my profs always said: "Wer misst, misst Mist!" (Which is fun in German ;) [Translation](https://dict.leo.org/forum/viewUnsolvedquery.php?idForum=1&idThread=440219&lp=ende&lang=en)

**Reply by abesamma** [reply]
That's a fun quote indeed!

**Reply by clutterstack** [reply]
Tangentially related:  "As soon as you have two thermometers, you no longer know the temperature" -- well-used in condensed matter physics

**Comment by pmario** [comment]
@kookma ... Is there a special reason, why it's important for you?

**Comment by clutterstack** [comment]
It feels as though TiddlyWiki can do a lot more than it can do fast enough, so to speak. So while the answer may be "no, there's not much we can speed up," or "certainly not without a messy rewrite" it's an interesting question for those of us not versed in the core.

Obviously any app will have its performance limitations. And if I want scripting on top of scripting, I'll have to accept that the constraints will be tighter.

It was an interesting revelation to me, when I started making pet plugins, that WikiText can do so much that I have to consider how to do it in the right way (or not in the wrongest way, at least), and to show restraint in how much I ask it to do at once (filters, listeners, even css).

@abesamma Point taken that TW is an application. Plenty of Electron apps are constantly disparaged for slowness, yet super-popular because usable and useful. Fit for purpose.

**Reply by abesamma** [reply]
@clutterstack VS code is the most useful of them all (i.e. electron apps) :) and hats off to the VS code team for improving it massively.

**Reply by clutterstack** [reply]
@abesamma Yes, I use VS code all the time. I'm very happy with it on Mac and Linux. No idea how well it works on Microsoft's own OS. Haha.

**Comment by kookma** [comment]
Thank you all for your inputs! Yes, TW is app and that is why it cannot have the performance of a static page!
So, may be Lighthouse is not the correct tools to measure the performance of an app like TW, but I was curious to see if we can boost the performance and quality of code!

**Reply by abesamma** [reply]
> 
> 
> Thank you all for your inputs! Yes, TW is app and that is why it cannot have the performance of a static page!
> So, may be Lighthouse is the not the correct the tools to measure the performance of an app like TW, but I was curious to see if we can boost the performance and quality of code!

Lighthouse isn't completely useless because as it correctly shows, TW5 isn't a PWA but it is useful for it to be IMO. I've actually experimented with this in Tiddlywiki's hosted at Maarfapad dot com, and it is very useful to just open TiddlyWiki like a normal app online and offline thanks to a separate service worker file.

Improving the SEO score is also important. There are many tools online that can help to improve the meta tags in a few minutes so this could be a good thing to look into.

**Comment by pmario** [comment]
Yea, Lighthouse does show us some weaknesses, that can be improved. ... But imo it wont be on the performance side of things. ... But we could have a closer look, why LH thinks tiddlywiki.com needs 12 seconds to start, which isn't true. ... At least for me
